webpack/assets/javascripts/react_app/components/bookmarks/form/index.js

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import { reduxForm } from 'redux-form';
+import { connect } from 'react-redux';
+import { required, length } from 'redux-form-validators';
+import Form from '../../common/forms/Form';
```

Comment:
I don't see a reason to use this `Form` component,
it doesn't give such a benefit and takes away flexibility.

* Submit and Cancel buttons should be in the `<Modal.Footer>`
* Submit and Cancel buttons should be in reverse order when inside `<Modal>`
* You should use `react-bootstrap`/`patternfly-react` `<Form horizontal></Form>`
* The form has some background that shouldn't be exist

http://www.patternfly.org/pattern-library/forms-and-controls/buttons-on-forms/
http://rawgit.com/sharvit/patternfly-react/storybook/forms/index.html?knob-Show%20Modal=true&knob-Show%20Feedback%20Icon=true&knob-Show%20Labels=false&knob-Show%20Help=true&knob-Show%20Loading=false&knob-Disabled=false&knob-Label=Danger%20Will%20Robinson%21&selectedKind=Forms&selectedStory=Modal%20Form&full=0&down=1&left=1&panelRight=0&downPanel=storybooks%2Fstorybook-addon-knobs

Reply:
@sharvit thanks, comments below:
> I don't see a reason to use this Form component, it doesn't give such a benefit and takes away flexibility.

One of the goals of using this, is to reduce the differences across forms, if you look at the implementation you would see its something pretty trivial to copy paste, and give us the ability to change the internal implementation if like in the future without changing all the forms.

> Submit and Cancel buttons should be in the <Modal.Footer>
True, I've noticed it as well, but it didnt seem to make a significant difference, and IMHO the point above is more important.

> Submit and Cancel buttons should be in reverse order when inside <Modal>
Why?

> You should use react-bootstrap/patternfly-react <Form horizontal></Form>
Yeah, that was created prior... I'll have a look.

> The form has some background that shouldn't be exist
?

Reply:
> One of the goals of using this, is to reduce the differences across forms, if you look at the implementation you would see its something pretty trivial to copy paste, and give us the ability to change the internal implementation if like in the future without changing all the forms.

The form component doesn't do much, just adding buttons and errors to your form.
I think that in forms, the right way to reduce differences is to have components that integrate well with each other instead of having 1 component that handles everything.
For example, in some cases you would want to use different buttons.

>> Submit and Cancel buttons should be in the <Modal.Footer>
>
> True, I've noticed it as well, but it didnt seem to make a significant difference, and IMHO the point above is more important.
>
>> Submit and Cancel buttons should be in reverse order when inside
>
> Why?

Only because patternfly documented it:
http://www.patternfly.org/pattern-library/forms-and-controls/buttons-on-forms/

>> You should use react-bootstrap/patternfly-react
>
> I'll have a look once its part of pf release?

Ok, but it is possible to use it from `react-bootstrap`. Then when pf-react will merge it will just be a matter of replacing the import statements.

>> The form has some background that shouldn't be exist
>
> ? do you mean the well css class?

Yes

Reply:
> Only because patternfly documented it:
> http://www.patternfly.org/pattern-library/forms-and-controls/buttons-on-forms/

I believe I remember hearing that they (or someone else) did some research and found that confirmation buttons on the right was more common and user friendly (even though we deal heavily in linux and linux tends to have them on the left).

Reply:
> > ? do you mean the well css class?
> 
> Yes

Can we please take this opportunity to remove the `.well`?